src/3d/shaders/phongDataDefined.vert

```diff
@@ -33,11 +33,10 @@ void main()
 
 
     // colors defined data
-    vs_out.ambient=vec4(dataDefinedAmbiantColor,1.0);
-    vs_out.diffuse=vec4(dataDefinedDiffuseColor,1.0);
-    vs_out.specular=vec4(dataDefinedSpecularColor,1.0);
+    vs_out.ambient=dataDefinedAmbiantColor;
+    vs_out.diffuse=dataDefinedDiffuseColor;
+    vs_out.specular=dataDefinedSpecularColor;
```

Comment:
Are you sure this is correct to apply opacity to all the colors here? Isn't that going to end up with incorrect color mixing? Shouldn't we apply opacity to the final color at the very end of the fragment shader?

Reply:
In fact, `phongFunction` only takes into account the rgb components of the colors and then applies the opacity of the diffuse color to the final color. Therefore, there is no risk of incorrect color mixing, this is already handled.

This means, that one only need to change `vs_out.diffuse` to have the expected result. However, it seems odd (to me) to change only one color. What do you prefer?